modules/xfem/src/userobjects/RectangleCutUserObject.C

```diff
@@ -21,36 +21,38 @@ validParams<RectangleCutUserObject>()
   InputParameters params = validParams<GeometricCut3DUserObject>();
 
   // Add required parameters
-  params.addRequiredParam<std::vector<Real>>("cut_data",
-                                             "Vector of Real values providing cut information");
+  params.addRequiredParam<Point>(
+      "vertex_point1", "Coordinates of the first of four vertices defining the rectangular cut");
+  params.addRequiredParam<Point>(
+      "vertex_point2", "Coordinates of the second of four vertices defining the rectangular cut");
+  params.addRequiredParam<Point>(
+      "vertex_point3", "Coordinates of the third of four vertices defining the rectangular cut");
+  params.addRequiredParam<Point>(
+      "vertex_point4", "Coordinates of the last of four vertices defining the rectangular cut");
   // Class description
   params.addClassDescription("Creates a UserObject for planar cuts on 3D meshes for XFEM");
   // Return the parameters
   return params;
 }
 
 RectangleCutUserObject::RectangleCutUserObject(const InputParameters & parameters)
-  : GeometricCut3DUserObject(parameters), _cut_data(getParam<std::vector<Real>>("cut_data"))
+  : GeometricCut3DUserObject(parameters),
+    _vertex_point1(getParam<Point>("vertex_point1")),
+    _vertex_point2(getParam<Point>("vertex_point2")),
+    _vertex_point3(getParam<Point>("vertex_point3")),
+    _vertex_point4(getParam<Point>("vertex_point4"))
 {
-  // Set up constant parameters
-  const int cut_data_len = 12;
-  const int num_vertices = 4;
-
-  // Throw error if length of cut_data is incorrect
-  if (_cut_data.size() != cut_data_len)
-    mooseError("Length of RectangleCutUserObject cut_data must be 12");
-
   // Assign cut_data to vars used to construct cuts
-  _vertices.push_back(Point(_cut_data[0], _cut_data[1], _cut_data[2]));
-  _vertices.push_back(Point(_cut_data[3], _cut_data[4], _cut_data[5]));
-  _vertices.push_back(Point(_cut_data[6], _cut_data[7], _cut_data[8]));
-  _vertices.push_back(Point(_cut_data[9], _cut_data[10], _cut_data[11]));
+  _vertices.push_back(_vertex_point1);
+  _vertices.push_back(_vertex_point2);
+  _vertices.push_back(_vertex_point3);
+  _vertices.push_back(_vertex_point4);
 
-  for (unsigned int i = 0; i < num_vertices; ++i)
+  for (unsigned int i = 0; i < _vertices.size(); ++i)
     _center += _vertices[i];
   _center *= 0.25;
```

Comment:
Magic number? I guess we are assuming rectangle here. It would be nice if this was a constant.

Reply:
how about `_center *= 1. / _vertices.size()`?